src/peergos/shared/user/FileUtil.java

```diff
@@ -0,0 +1,53 @@
+package peergos.shared.user;
+
+import peergos.shared.Crypto;
+import peergos.shared.NetworkAccess;
+import peergos.shared.user.fs.FileWrapper;
+import peergos.shared.util.Futures;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+
+public class FileUtil {
+
+
+    public static String getFilename(Path p) {
+        return p.getName(p.getNameCount() - 1).toString();
+    }
+
+    private static List<String> toList(Path p) {
+        return Arrays.asList(toRelative(p).toString().split("/"));
+    }
+
+    public static Path canonicalise(Path p) {
+        return p.isAbsolute() ? p : Paths.get("/").resolve(p);
+    }
+
+    public static Path toRelative(Path p) {
+        return p.isAbsolute() ? Paths.get(p.toString().substring(1)) : p;
+    }
+
+    public static CompletableFuture<FileWrapper> getOrMkdir(FileWrapper parent, String dirName, Crypto crypto, NetworkAccess network) {
```

Comment:
getOrMkdir variants can be removed because it exists on FileWrapper now

Reply:
done